Fix publish workflow

@WangGithub0 I tried running your publish workflow.  I created a new Release and Tag:

<img width="1304" alt="Screenshot 2024-01-24 at 4 39 02 PM" src="https://github.com/tarasglek/typescript2openai/assets/427398/611a74b5-c2e4-44c0-a5f9-29d2096e4703">

I also had it Generate Release Notes for me.

The action ran, see https://github.com/tarasglek/typescript2openai/actions/runs/7646947777/job/20836855307, but it failed:

```
ERR_PNPM_GIT_UNCLEAN  Unclean working tree. Commit or stash changes first.

If you want to disable Git checks on publish, set the "git-checks" setting to "false", or run again with "--no-git-checks".
```

This is due to the git working dir not being clean, and it's not a problem in CI.  You can fix it by updating your code here:

https://github.com/tarasglek/typescript2openai/blob/main/.github/workflows/npm-publish.yml#L25

You want to use `--no-git-checks`, see https://pnpm.io/cli/publish#--no-git-checks.

--- a/.github/workflows/npm-publish.yml
+++ b/.github/workflows/npm-publish.yml
@@ -22,4 +22,4 @@ jobs:
           run_install: true
       - run: pnpm build
       - run: pnpm test
-      - run: pnpm publish
+      - run: pnpm publish --no-git-checks

